datafusion/core/benches/sql_planner_extended.rs

```diff
@@ -0,0 +1,225 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::array::{ArrayRef, RecordBatch};
+use arrow_schema::DataType;
+use arrow_schema::TimeUnit::Nanosecond;
+use criterion::{criterion_group, criterion_main, Criterion};
+use datafusion::prelude::{DataFrame, SessionContext};
+use datafusion_catalog::MemTable;
+use datafusion_common::ScalarValue;
+use datafusion_expr::Expr::Literal;
+use datafusion_expr::{cast, col, lit, not, try_cast, when};
+use datafusion_functions::expr_fn::{
+    btrim, length, regexp_like, regexp_replace, to_timestamp, upper,
+};
+use std::ops::Rem;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+/// Registers a table like this:
```

Comment:
Maybe we could add some context to this benchmark that explains the reproducer is very slow and designed to cover certain planning cases, referencing

https://github.com/apache/datafusion/issues/17261